docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+![submit_pr_remote_tracking_3.png](./imgs/submit_pr_remote_tracking_3.png)
+
+
+Return to your terminal window and type :
+
+```
+git remote add upstream git@github.com:strongloop/loopback-next.git
+```
+
+then type :
+
+```
+git remote
+```
+
+there should now be two remotes : **origin** and **upstream**.
+
+![submit_pr_remote_tracking_4.png](./imgs/submit_pr_remote_tracking_4.png)
+
+Typing
+
+```
+git remote -v
+```
+
+provides more details.
+
+![submit_pr_remote_tracking_5.png](./imgs/submit_pr_remote_tracking_5.png)
+
+### 8. Install dependencies
+
+The `loopback-next` has several package dependencies.
+
+Type `npm install` to install them.
+
+![submit_pr_npm_install_1.png](./imgs/submit_pr_npm_install_1.png)
+
+
+### 9. Do some work in your local feature branch directory
+
+Whether you are contributing to code or documentation, be sure to make all your changes inside in the local feature branch directory.
+
+Be sure to read, understand, and abide by the instructions provided in [Contributing code](https://loopback.io/doc/en/contrib/code-contrib.html), [Contributing to docs](https://loopback.io/doc/en/contrib/doc-contrib.html), and
+[Contributing to LoopBack](https://github.com/strongloop/loopback-next/blob/master/docs/CONTRIBUTING.md)
+
+### 10. Verifying your code changes
+
+Before `staging your changes` (the next step), be sure your files are free of formatting, syntax, and logical/execution errors.
+
+To flush out any **formatting** issues in code, run `npm run lint`. If there are issues, run `npm run lint:fix`.
+
+To flush out any **syntax** issues in code, run `npm run build`. If there are issues, fix them, and run `npm run build` again. Repeat until all issues are gone.
+
+To flush out any **logical/execution** issues in code, run `npm run test`. This will run the existing mocha test cases, and any new test cases you have added. If there are issues, fix them, run `npm build`, and run `npm run test` again. Repeat until all issues are gone.
+
+### 11. Verifying your documentation changes
+
+To flush out any **formatting** issues in documentation, run `npm run lint`. If there are issues, run `npm run lint:fix`.
+
+Documentation written in **Markdown** format needs to be compiled into **HTML** files. To flush out any layout or substitution issues ( from using 'includes' for example), run `npm run build:site`. If there are any issues, fix them and run `npm run build:site` again.
+Then `cd sandbox/loopback.io/` and run `npm start` to start a local web server that hosts the HTML documentation.
+
+![submit_pr_check_local_docs_1.png](./imgs/submit_pr_check_local_docs_1.png)
+
+
+In your browser, navigate to the specified server address, click on `Docs`, click on `Loopback 4`, and verify that your document changes appear and are well-rendered.
+
+Press `Ctrl+C` to stop the local documentation server.
+
+
+
+
+### 12. Staging your changes
```

Comment:
We should refer to git/github docs to teach folks how to stage/commit/push.

Reply:
I was asked to use https://www.nearform.com/blog/getting-into-open-source-for-the-first-time/ as inspiration for my document, so that's the "flavour" I followed. We can certainly provide a link to github docs in the References. Like the document you mentioned: https://help.github.com/categories/collaborating-with-issues-and-pull-requests/.